D1673/P1673.md

```diff
@@ -6069,97 +6069,103 @@ public:
     using layout_type = layout_blas_packed<Triangle, StorageOrder>;
 
-  private:
-    Extents @_the-extents_@{}; // exposition only
-
-  public:
     constexpr mapping() noexcept = default;
     constexpr mapping(const mapping&) noexcept = default;
-    constexpr mapping(const extents_type& e) noexcept;
+    constexpr mapping(const extents_type&) noexcept;
     template<class OtherExtents>
       constexpr explicit(! is_convertible_v<OtherExtents, extents_type>)
         mapping(const mapping<OtherExtents>& other) noexcept;
 
     constexpr mapping& operator=(const mapping&) noexcept = default;
 
-    constexpr extents_type extents() const noexcept { return @_the-extents_@; }
+    constexpr const extents_type& extents() const noexcept { return @_extents\__@; }
 
-    constexpr size_type required_span_size() const noexcept;
+    constexpr index_type required_span_size() const noexcept;
 
     template<class... Indices>
       constexpr index_type operator() (Indices...) const noexcept;
 
-    static constexpr bool is_always_unique() {
+    static constexpr bool is_always_unique() noexcept {
       return extents_type::static_extent(0) != dynamic_extent &&
         extents_type::static_extent(0) < 2;
     }
-    static constexpr bool is_always_exhaustive() { return true; }
-    static constexpr bool is_always_strided() {
+    static constexpr bool is_always_exhaustive() noexcept { return true; }
+    static constexpr bool is_always_strided() noexcept {
       return extents_type::static_extent(0) != dynamic_extent &&
         extents_type::static_extent(0) < 2;
     }
 
     constexpr bool is_unique() const noexcept {
-      return extent(0) < 2;
+      return @_extents\__@.extent(0) < 2;
     }
     constexpr bool is_exhaustive() const noexcept { return true; }
     constexpr bool is_strided() const noexcept {
-      return extent(0) < 2;
+      return @_extents\__@.extent(0) < 2;
     }
 
     constexpr index_type stride(rank_type) const noexcept;
 
     template<class OtherExtents>
       friend constexpr bool
         operator==(const mapping&, const mapping<OtherExtents>&) noexcept;
+
+  private:
+      extents_type @_extents\__@{}; // exposition only
+  };
 };
 ```
 
-[4]{.pnum} *Constraints:*
+[4]{.pnum} *Mandates:*
 
   * [4.1]{.pnum} `Triangle` is either `upper_triangle_t`
-      or `lower_triangle_t`.
+      or `lower_triangle_t`,
 
   * [4.2]{.pnum} `StorageOrder` is either `column_major_t`
-      or `row_major_t`.
+      or `row_major_t`,
+
+  * [4.3]{.pnum} `Extents` is a specialization of `extents`,
 
-  * [4.3]{.pnum} `Extents` is a specialization of `extents`.
+  * [4.4]{.pnum} `Extents::rank()` equals 2, and
 
-  * [4.4]{.pnum} `Extents::rank()` equals 2.
+  * [4.5]{.pnum} one of `extents_type::static_extent(0) == dynamic_extent`,
+                        `extents_type::static_extent(1) == dynamic_extent`, or
+                  `extents_type::static_extent(r1) == extents_type::static_extent(1))` is `true`.
+
```

Comment:
Do we also need the Mandates from [mdspan.layout.left.overview] 4?

```suggestion
  * [4.5]{.pnum} one of `extents_type::static_extent(0) == dynamic_extent`,
                        `extents_type::static_extent(1) == dynamic_extent`, or
                  `extents_type::static_extent(r1) == extents_type::static_extent(1))` is `true`, and

  * [4.6]{.pnum} If `Extents​::​rank_dynamic() == 0` is true, then the size of the multidimensional index space `Extents()` is representable as a value of type `typename Extents​::​index_type`.
```

Reply:
thought about it but technically we only need half of it?